backend/src/packages/orders/order.service.ts

```diff
@@ -0,0 +1,181 @@
+import {
+  type UserEntityObjectWithGroupT,
+  type ValueOf,
+  HttpCode,
+  HttpMessage,
+} from 'shared/build/index.js';
+
+import { HttpError } from '~/libs/exceptions/exceptions.js';
+import { type IService } from '~/libs/interfaces/interfaces.js';
+
+import { OrderStatus } from './libs/enums/enums.js';
+import {
+  type OrderCreateRequestDto,
+  type OrderCreateResponseDto,
+  type OrderEntity as OrderEntityT,
+  type OrderUpdateResponseDto,
+} from './libs/types/types.js';
+import { OrderEntity } from './order.entity.js';
+import { type OrderRepository } from './order.repository.js';
+
+class OrderService implements Omit<IService, 'delete' | 'findById'> {
+  private orderRepository: OrderRepository;
+
+  private errors: Record<
+    string,
+    {
+      status: ValueOf<typeof HttpCode>;
+      message: string;
+    }
+  >;
```

Comment:
Im not sure we need this private field, we can just throw errors if something goes wrong

Reply:
replace with custom error for orders